server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -259,9 +177,15 @@ public ClusterState execute(ClusterState currentState) {
                     decommissionAttributeMetadata.decommissionAttribute(),
                     decommissionStatus
                 );
-                return ClusterState.builder(currentState)
+                ClusterState newState = ClusterState.builder(currentState)
                     .metadata(Metadata.builder(currentState.metadata()).decommissionAttributeMetadata(decommissionAttributeMetadata))
                     .build();
+
+                // For terminal status we will go ahead and clear any exclusion that was added as part of decommission action
+                if (decommissionStatus.equals(DecommissionStatus.SUCCESSFUL) || decommissionStatus.equals(DecommissionStatus.FAILED)) {
+                    newState = clearExclusionsAndGetState(newState);
```

Comment:
Lets ensure we call this out in the documentation

Reply:
Ack